src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,54 @@
 package mate.academy.spring.config;
 
+import static mate.academy.spring.model.Role.RoleName.ADMIN;
+import static mate.academy.spring.model.Role.RoleName.USER;
+
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers(HttpMethod.POST, "/register").permitAll()
+
+                .antMatchers(HttpMethod.GET, "/cinema-halls").hasAnyRole(ADMIN.name(), USER.name())
+                .antMatchers(HttpMethod.GET, "/movies").hasAnyRole(ADMIN.name(), USER.name())
+                .antMatchers(HttpMethod.GET, "/movie-sessions/available").hasAnyRole(ADMIN.name(),
+                        USER.name())
```

Comment:
You can write this url through a comma.
```suggestion
                .antMatchers(HttpMethod.GET, "/cinema-halls/", "/movies/,/movie-sessions/available").hasAnyRole(ADMIN.name(), USER.name())
              
```

Reply:
I think it is more readable. And it is also grouped by roles, so in case I need to modify smth I will find the role or the url faster